python/tests/erdos_to_ros_test.py

```diff
@@ -0,0 +1,169 @@
+NUM_RECEIVED = Value("i", 0)
+TEST_FAILED = Value("i", 0)
+
+
+class SendOp(erdos.Operator):
+    """Generates and sends a stream of Erdos messages.
+    Args:
+        write_stream: Stream to which messages are written
+        messages: List of Python primitives or objects that
+            are sent as Erdos messages in the write stream.
+    """
+    def __init__(self, write_stream, messages):
+        self.write_stream = write_stream
+        self.messages = messages
+        self.coord = 0
+
+    @staticmethod
+    def connect():
+        return [erdos.WriteStream()]
+
+    def run(self):
+        for raw_msg in self.messages:
+            msg = erdos.Message(erdos.Timestamp(coordinates=[self.coord]),
+                                raw_msg)
+            print("SendOp: sending {msg}".format(msg=msg))
+            self.write_stream.send(msg)
+            self.coord += 1
+            time.sleep(0.25)
+
+
+class ErdosToRosOp(erdos.Operator):
+    """Converts Erdos messages to ROS messages.
+    Args:
+        publish_stream (:py:class:`erdos.ReadStream`): Stream on which the
+            operator receives control commands.
+        ros_msg_type: Type used to publish to ros messages
+        rostopic: rostopic to publish messages to
+        node_name: Ros node name
+        func: Callback function that converts a single ros_msg to an erdos_msg
+    """
+    def __init__(self, publish_stream, ros_msg_type, rostopic, node_name,
+                 func):
+        self.logger = erdos.utils.setup_logging(self.config.name,
+                                                self.config.log_file_name)
+        self.publish_stream = publish_stream
+        self.publish_stream.add_callback(self.on_erdos_msg)
+        self.ros_msg_type = ros_msg_type
+        self.rostopic = rostopic
+        self.node_name = node_name
+        self.func = func
+        self.ros_pub = None
+
+    @staticmethod
+    def connect(read_streams):
+        return []
+
+    def on_erdos_msg(self, msg):
+        """
+        Callback for publish_stream that converts
+        erdos messages to ros messages and then
+        published them.
+
+        msg: Erdos message that will be converted
+            to ros message and then published.
+        """
+
+        new_ros_msg = self.func(msg)
+        print("Publishing: " + str(new_ros_msg))
+        try:
+            self.ros_pub.publish(new_ros_msg)
+        except (NameError, AttributeError) as err:
+            self.logger.debug(
+                "Ros publisher was not defined, \
+                possibly because run() was not called: %s", err)
+        except (ROSException, ROSSerializationException) as err:
+            self.logger.debug("The erdos msg couldn't publish to ros: %s", err)
+
+    def run(self):
+        # Initialize a publisher
+        self.ros_pub = rospy.Publisher(self.rostopic,
+                                       self.ros_msg_type,
+                                       queue_size=10)
```

Comment:
The `queue_size` should not be set to a default since its choice has repercussions on the performance.